zio-http/src/main/scala/zio/http/api/MiddlewareSpec.scala

```diff
@@ -16,6 +17,9 @@ final case class MiddlewareSpec[MiddlewareIn, MiddlewareOut](
   ): MiddlewareSpec[inCombiner.Out, outCombiner.Out] =
     MiddlewareSpec(self.middlewareIn ++ that.middlewareIn, self.middlewareOut ++ that.middlewareOut)
 
+  def handle[R, E](f: MiddlewareIn => ZIO[R, E, MiddlewareOut]): Middleware[R, E, MiddlewareIn, MiddlewareOut] =
```

Comment:
I like this for symmetry. Maybe they should both be called `implement`. 

e.g.

```scala
myApi.implement(...)

myMiddlewareSpec.implement(...)
```

The weird thing is that `Middleware` doesn't store a `MiddlewareSpec`, while `Service` stores an `API`. Maybe that should change?

Reply:
OTOH `handle` is shorter even if its use for middleware is a bit awkward.

Reply:
Probably will use `handle` for Service and `implement` for middleware.?